examples/dao/test/failing-path.js

```diff
@@ -0,0 +1,585 @@
+      // set current time when "voting is active"
+      ctx.runtime.setRoundAndTimestamp(10, votingStart + 30);
+      assert.throws(
+        () => ctx.executeTx(executeProposalTx),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should reject execution if proposal.yes < min_support', () => {
+      // register votes by A (< min_support)
+      ctx.depositVoteToken(ctx.voterA, ctx.depositLsig, minSupport - 1);
+      ctx.vote(ctx.voterA, Vote.YES, ctx.proposalLsigAcc);
+
+      assert.throws(
+        () => ctx.executeTx(executeProposalTx),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should reject execution if proposal.yes > min_support, but vote.no > vote.yes', () => {
+      // register 2 more votes by A (> min_support)
+      ctx.depositVoteToken(ctx.voterA, ctx.depositLsig, minSupport + 1);
+      ctx.vote(ctx.voterA, Vote.YES, ctx.proposalLsigAcc);
+
+      // register votes by voterB (where NO votes > YES votes by voterA)
+      ctx.depositVoteToken(ctx.voterB, ctx.depositLsig, minSupport + 5);
+      ctx.vote(ctx.voterB, Vote.NO, ctx.proposalLsigAcc);
+
+      assert.throws(
+        () => ctx.executeTx(executeProposalTx),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should reject execution if proposal already executed', () => {
+      // register yes votes
+      ctx.depositVoteToken(ctx.voterA, ctx.depositLsig, minSupport + 1);
+      ctx.vote(ctx.voterA, Vote.YES, ctx.proposalLsigAcc);
+
+      // set executed == true
+      ctx.proposalLsigAcc.setLocalState(ctx.daoAppID, 'executed', 1n);
+
+      assert.throws(
+        () => ctx.executeTx(executeProposalTx),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should reject execution if 2nd transaction not included', () => {
+      // register yes votes
+      ctx.depositVoteToken(ctx.voterA, ctx.depositLsig, minSupport + 1);
+      ctx.vote(ctx.voterA, Vote.YES, ctx.proposalLsigAcc);
+
+      assert.throws(
+        () => ctx.executeTx({ ...executeProposalTx[0] }),
+        INDEX_OUT_OF_BOUND_ERR
+      );
+    });
+
+    it('should reject execution if 2nd transaction is not per proposal instructions', () => {
+      // register yes votes
+      ctx.depositVoteToken(ctx.voterA, ctx.depositLsig, minSupport + 1);
+      ctx.vote(ctx.voterA, Vote.YES, ctx.proposalLsigAcc);
+
+      // amount is wrong
+      assert.throws(
+        () => ctx.executeTx([
+          { ...executeProposalTx[0] },
+          { ...executeProposalTx[1], amountMicroAlgos: 10n }
+        ]),
+        RUNTIME_ERR1009
+      );
+
+      // recipient is wrong
+      assert.throws(
+        () => ctx.executeTx([
+          { ...executeProposalTx[0] },
+          { ...executeProposalTx[1], toAccountAddr: ctx.voterA.address }
+        ]),
+        RUNTIME_ERR1009
+      );
+
+      // from_account is wrong
+      assert.throws(
+        () => ctx.executeTx([
+          { ...executeProposalTx[0] },
+          { ...executeProposalTx[1], fromAccountAddr: ctx.voterA.address }
```

Comment:
Probably something is wrong her: everyone should be able to execute transaction.

Reply:
> everyone should be able to execute transaction

yeah, that's *tx0*, which is the first tx (call to DAO app). Here we change the "from account" of transaction as per proposal instructions (*tx1*). And should fail if from account address is different from the one recorded in original proposal

Reply:
ah, let's add a comment to make it easier for the reader. Thanks.